sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs

```diff
@@ -5702,5 +6126,70 @@ public virtual Uri GenerateSasUri(ShareSasBuilder builder)
             return sasUri.ToUri();
         }
         #endregion
+
+        #region PartitionedDownloader
+        internal PartitionedDownloader<ShareFileRequestConditions, ShareFileDownloadInfo> GetPartitionedDownloader(
+            StorageTransferOptions transferOptions,
+            string operationName = null)
+            => new PartitionedDownloader<ShareFileRequestConditions, ShareFileDownloadInfo>(
+                GetPartitionedDownloaderBehaviors(this),
+                transferOptions,
+                operationName);
+
+        internal static PartitionedDownloader<ShareFileRequestConditions, ShareFileDownloadInfo>.Behaviors GetPartitionedDownloaderBehaviors(ShareFileClient client)
+        {
+            return new PartitionedDownloader<ShareFileRequestConditions, ShareFileDownloadInfo>.Behaviors
+            {
+                SingleDownload = async (range, args, rangeGetContentHash, async, cancellationToken)
+                    => await client.DownloadInternal(
+                        range,
+                        rangeGetContentHash,
+                        args,
+                        async,
+                        cancellationToken).ConfigureAwait(false),
+                CopyToAsync = async (result, destination, cancellationToken)
+                    =>
+                {
+                    using Stream source = result.Content;
+
+                    await source.CopyToAsync(
+                        destination,
+                        Constants.DefaultDownloadCopyBufferSize,
+                        cancellationToken)
+                        .ConfigureAwait(false);
+                },
+                CopyTo = (result, destination, cancellationToken)
+                    =>
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    result.Content.CopyTo(destination, Constants.DefaultDownloadCopyBufferSize);
+                    result.Content.Dispose();
+                },
```

Comment:
This doesn't seem to be much different from blobs. Should we move it to downloader?

This is how we get access to stream in our OpenRead family.

https://github.com/Azure/azure-sdk-for-net/blob/c87f56cd6184c318cd373ab7302618261d193170/sdk/storage/Azure.Storage.Common/src/Shared/LazyLoadingReadOnlyStream.cs#L198-L200

Reply:
Huh... I had pulled it out from the downloader because there was a conflict arising from the response value type being generic, but that issue isn't happening now. Fixed, will be in next push.